core/store/packer.rs

```diff
@@ -101,6 +102,8 @@ impl Packer {
             .build()
             .unwrap();
 
+        pkg.update(target.clone()).await?;
```

Comment:
The `update` function doesn't really need ownership of this value, so we should pass in a `&target` and refactor it accordingly.